### PR TITLE
[NL Next] A few fulfillment fixes

### PR DESCRIPTION
**Pull request by pradh**
1. Downgrade single point time-series to bar charts
2. Open up topics in correlation, and include multiple SVs from current utterance and past context
3. Show contained-in maps for peer-group SVs
4. Sprinkle more debug logs

This improves on Jan script, in these ways:

(1) [agriculture in california] contained single-dot timeline charts
Screenshot 2023-02-02 at 12 11 47 PM

(2) There used to be only one correlation chart
Screenshot 2023-02-02 at 12 12 22 PM

### CODE REVIEW

File: server/lib/nl/fulfillment/base.py [hunk 1]
```diff
@@ -249,3 +253,38 @@ def _svg_or_topic_to_svs(state: PopulateState, sv: str,
     return charts
 
   return []
+
+
+def open_topics_ordered(svs: List[str]) -> List[str]:
+  opened_svs = []
+  sv_set = set()
+  for rank, sv in enumerate(svs):
```
Comment (jehangiramjad) [inline server/lib/nl/fulfillment/base.py:261]: is `svs` expected to be ordered in some way to have `rank` make sense? If so, please add a comment to that effect.

File: server/lib/nl/fulfillment/base.py [hunk 2]
```diff
@@ -249,3 +253,38 @@ def _svg_or_topic_to_svs(state: PopulateState, sv: str,
...
+    for sv in _open_topic_in_var(sv, rank):
```
Comment (jehangiramjad) [inline server/lib/nl/fulfillment/base.py:262]: both the inner and outer loops are using the same var name `sv`?

File: server/lib/nl/fulfillment/base.py [hunk 3]
```diff
@@ -249,3 +253,38 @@ def _svg_or_topic_to_svs(state: PopulateState, sv: str,
...
+      if sv not in sv_set:
```
Comment (jehangiramjad) [inline server/lib/nl/fulfillment/base.py:263]: why use both a set and a list? could we not just use the set and return list(sv_set)?

File: server/lib/nl/fulfillment/base.py [hunk 4]
```diff
@@ -249,3 +253,38 @@ def _svg_or_topic_to_svs(state: PopulateState, sv: str,
...
+        opened_svs.append(sv)
+        sv_set.add(sv)
+  return opened_svs
+
+
+def _open_topic_in_var(sv: str, rank: int) -> List[str]:
+  if utils.is_sv(sv):
+    return [sv]
+  if utils.is_topic(sv):
```
Comment (jehangiramjad) [inline server/lib/nl/fulfillment/base.py:272]: Why is nothing being returned from this conditional path?

File: server/lib/nl/utils.py [hunk 5]
```diff
@@ -170,6 +170,21 @@ def rank_svs_by_latest_value(place: str, svs: List[str],
...
+                                 all_facets=False)
+  for _, place_data in series_data['data'].items():
+    if place not in place_data:
+      continue
+    series = place_data[place]['series']
+    if len(series) < 2:
+      logging.info('Found single data point series in %s - %s', place,
+                   ', '.join(svs))
+      return True
+  return False
```
Comment (jehangiramjad) [inline server/lib/nl/utils.py:185]: Just checking the logic: a series with no points is also false because it does NOT have a single data point?

**Review by jehangiramjad — COMMENTED** [review]
Thanks. Some questions. Haven't reviewed the tests carefully though

**Review by jehangiramjad — APPROVED** [review]
Approving to unblock

File: server/lib/nl/fulfillment/base.py (hunk 2 above)
Comment (pradh) [reply server/lib/nl/fulfillment/base.py:262]: good catch!

File: server/lib/nl/fulfillment/base.py (hunk 3 above)
Comment (pradh) [reply server/lib/nl/fulfillment/base.py:263]: Because we want it ordered.

File: server/lib/nl/fulfillment/base.py (hunk 4 above)
Comment (pradh) [reply server/lib/nl/fulfillment/base.py:272]: Great eyes!  I just fixed the bug.

File: server/lib/nl/fulfillment/base.py (hunk 1 above)
Comment (pradh) [reply server/lib/nl/fulfillment/base.py:261]: Added comments

**Pull request merged**